### ISSUE TITLE
libSBML

### ISSUE THREAD

**Issue by dlill** [dkaschek/dMod#56]
Hey, I just tried to build dMod on may Laptop, this is what I get:

```
==> R CMD INSTALL --no-multiarch --with-keep.source dMod

* installing to library ‘/home/daniel/R/x86_64-pc-linux-gnu-library/3.6’
ERROR: dependency ‘libSBML’ is not available for package ‘dMod’
* removing ‘/home/daniel/R/x86_64-pc-linux-gnu-library/3.6/dMod’

Exited with status 1.
```

But this doesn't work either
```
> install.packages("libSBML")
Installing package into ‘/home/daniel/R/x86_64-pc-linux-gnu-library/3.6’
(as ‘lib’ is unspecified)
Warning message:
package ‘libSBML’ is not available (for R version 3.6.3) 
```

I think we should not depend on non-CRAN packages, or if we do, the installation procedure needs to be documented properly in the Readme.md 
What do you think?

**Comment by svenjakemmer** [comment]
Hey, if we want to include PEtab2dMod in dMod we need libSBML. It provides the core functions for the import of SBML models in R. That's why we included a documentation in the dMod Wiki how to install it: https://github.com/dkaschek/dMod/wiki/Support-for-PEtab

**Comment by marcusrosenblatt** [comment]
Hi @dlill,

@svenjakemmer wrote on this in [Support for PEtab](https://github.com/dkaschek/dMod/wiki/Support-for-PEtab)

I know it is not super nice to have to install packages in that way. Although, similarly we have it for cOde.

**Comment by marcusrosenblatt** [comment]
Frank Bergmann from the libSBML developer team said that they cannot bring the package to CRAN. But maybe one could ask them to provide it as a github repository such that installing via devtolls would work!?

**Comment by dlill** [comment]
@svenjakemmer can you put the short installation instruction including a download link also into the readme.md? That's where people will look first.
@marcusrosenblatt good Idea, the devtools thing. Apart from that, what do you mean with cOde? install.packages("cOde") works??

**Comment by marcusrosenblatt** [comment]
Yes, it works of course. But if one needs the current version 1.1.0 to be used with the development version of dMod, then you have to you devtools. Otherwise you install cOde 1.0.0.

Or did this change recently? Did not reinstall for about two weeks.

**Comment by marcusrosenblatt** [comment]
I asked Frank and they are planning to put it on github anyway during the next weeks. Installing via devtools should then also be easily doable.

For the moment, we can also use install_url()

**Comment by dlill** [comment]
Cool, sounds good. Can you then put the complete call to `install_url()` including the url into the readme.md?

**Comment by marcusrosenblatt** [comment]
Done for Linux. Windows installation like

`devtools::install_url("https://sourceforge.net/projects/sbml/files/libsbml/5.18.0/stable/Windows/64-bit/R%20interface/libSBML_5.18.0.zip")`

did not work for me immediately. Will be updated later.

**Comment by svenjakemmer** [comment]
As we had some issues with the installation file provided by the SBML community on external clusters we decided to excluded libSBML from the dependencies again.  I made a remark, that it is needed for the functionality of the SBML support with a link to installation instructions.
